test/token_holder/constructor.js

```diff
@@ -0,0 +1,91 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+const utils = require('../test_lib/utils.js');
+
+const TokenHolder = artifacts.require('TokenHolder');
+
+contract('TokenHolder::constructor', async () => {
+    contract('Negative testing for input parameters.', async (accounts) => {
+        it('Null branded token address.', async () => {
+            const required = 1;
+            const registeredWallet0 = accounts[0];
+
+            const wallets = [registeredWallet0];
+
+            const brandedTokenAddress = utils.NULL_ADDRESS;
+            const tokenRulesAddress = accounts[1];
+
+            await utils.expectRevert(
+                TokenHolder.new(
+                    brandedTokenAddress,
+                    tokenRulesAddress,
+                    wallets,
+                    required,
+                ),
+                'Branded token address cannot be null.',
+            );
+        });
+
+        it('Null token rules address.', async () => {
```

Comment:
As discussed in previous PR, It is more readable if "it" makes a proper statement. It's easy to read and understand what test case is going to do. We can take others opinion also but this we have been following for quite long. Its fine if we want to take it up as part of improvements and update all at once.

Here for e.g. it('Fails for Null token rules address.') 
Your thoughts?